### ISSUE TITLE
Libguides: Empty container element

### ISSUE THREAD

**Issue by rschwab** [UCSCLibrary/SiteImprove#45]
[Siteimprove link](https://my2.siteimprove.com/Accessibility/624596/77184128804/NextGen/Issue/1?conformance=&siteTargetIssueKinds=1,2&wcagVersion=21&pageSegments=&ruleName=sia-r68&ruleId=68&issueKind=1&tagFilterType=2&siteTarget.wcagVersion=21&siteTarget.conformanceLevels=0,1,3,4&siteTarget.issueKinds=1,2)

Too many pages to list here. One of the primary (only?) location of this error comes from Springshares code. Ticket is submitted.

**Comment by rschwab** [comment]
Ticket ID: 369530

Feb 07 2025, 03:07PM via System
Accessibility issue: Empty container elements
Hello Springshare!

There is an accessibility issue in Libguides menus that is breaking conformance with WCAG2.1 A. The problem is empty container elements, in this case the menu links on a multi-page guide. The UL element in the 's-lg-tabs-container' element has role="menu". To fix this issue, the LI elements within that list need to have role="menuitem".

See: https://www.w3.org/TR/wai-aria-1.1/#mustContain

**Comment by rschwab** [comment]
This issue is directly related to the "Role not inside the required context" issue, which is where the list items in a menu have defaulted to role="listitem" when they need role="menuitem"

**Comment by rschwab** [comment]
Springshare reply: 

Good news-- this issue was actually resolved back in 2016! This is likely a case of a custom template made before that update was implemented. Unfortunately, once a template has been customized, it can diverge from our default templates as those are automatically updated with any accessibility (or other) changes as needed. We do try to let folks using custom templates know about how to incorporate these changes, such as in these examples ([one](https://blog.springshare.com/2019/06/25/june-release/) and [two](https://blog.springshare.com/2023/04/12/our-ongoing-commitment-to-accessibility/)) from a blog update.

In order to resolve this issue in your custom template (I believe it's the "Top to Side" template), you will need to make a little code adjustment. 

    Head to Admin > Look & Feel > Page Layout > Guide
    In the Customize Guide Templates panel, load the tabbed layout template.
    Look through the code for this section, which starts about 30 lines from the top ([see screenshot](https://app.screencast.com/wgZbnlSyNOnxL)).
    Copy the s-lg-tabs-container section
    Load the Top to Side template and replace the s-lg-tabs-container in your custom template with the copied section from the System Sefault - Tab Layout template
    Save the Top to Side template.
    If not being resolved on the guides, current tabbed navigation guides may need to have their layout template re-saved, so that they use the updated template.
        Edit a guide
        Head to the Guide Layout menu
        Pick Guide Navigation Layout
        Select the tabbed layout template.
        Click save.

**Comment by JessWaggs** [comment]
@rschwab I'm not sure how to fix this one: https://my2.siteimprove.com/Inspector/624596/A11Y/Page?pageId=183324142320&impmd=D04AB39525FEB7C9C85D21DD5754A0C4&issue=cantTell+failed&wcag=twopointone&decision=dismissed+falsePositive+none#/sia-r68/failed/bmNsdWtVbS8=;YWNkMDYzNGY=

Can you take a look?

**Comment by JessWaggs** [comment]
Reported the error on embedded LibCal widgets to Springshare 7/2025

**Comment by rschwab** [comment]
> [@rschwab](https://github.com/rschwab) I'm not sure how to fix this one: https://my2.siteimprove.com/Inspector/624596/A11Y/Page?pageId=183324142320&impmd=D04AB39525FEB7C9C85D21DD5754A0C4&issue=cantTell+failed&wcag=twopointone&decision=dismissed+falsePositive+none#/sia-r68/failed/bmNsdWtVbS8=;YWNkMDYzNGY=
> 
> Can you take a look?

The problem was some leftover HTML in the box. I removed it and the issue is now fixed. It did make the space between the search box and the how-to box narrower - do you want me to add some padding back in?

**Comment by JessWaggs** [comment]
@rschwab  I think it looks fine like this. Thanks!
